task13/index.js

```diff
@@ -0,0 +1,96 @@
+const parent = document.getElementById("js-parent"); 
+
+const appendModalForParent = () => {
+    const modalPlace = document.createElement('section');
+    const modalText = document.createElement('p');
+    const closeBtn = document.createElement('input');
+    modalPlace.style.margin = '0 auto';
+    modalPlace.style.textAlign = "center";
+    modalPlace.style.width = '300px';
+    modalPlace.style.padding = '20px';
+    modalPlace.style.backgroundColor = "lightGray";
+    modalPlace.id = "modalPlace";
+    modalText.textContent = "クリックしてデータを取得します。";
+    closeBtn.style.margin = '0 auto';
+    closeBtn.style.display = "block";
+    closeBtn.id = "closeBtn";
+    closeBtn.type = "button";
+    closeBtn.value = "click";
+
+    parent.appendChild(modalPlace).insertBefore(modalText, modalPlace.firstChild).appendChild(closeBtn);
+};
+
+const appendButtonForParent = () => {
+    const button = document.createElement('input');
+    button.id = "button";
+    button.type = "button";
+    button.value = "click";
+    parent.appendChild(button);
+};
+appendButtonForParent();
+
+const openBtn = document.getElementById("button");
+openBtn.addEventListener("click", (e) => {
+    appendModalForParent();
+    e.target.remove();
+    modalClose();
+});
+
+const modalClose = () => {
+    const closeBtn = document.getElementById("closeBtn");
+    closeBtn.addEventListener("click", () => {
+        document.getElementById("modalPlace").remove();
+        init();
+    });
+};
+
+const getData = async () => {
+    const response = await fetch("https://jsondata.okiba.me/v1/json/3jhPW210812052923");
+    const data = response.json();
+    return data;
+};
+
+const loading = () => {
+    const loadingPlace = document.createElement("div");
+    const gif = document.createElement('img');
+    loadingPlace.id = "loadingPlace";
+    gif.src = "loading-circle.gif";
+    parent.appendChild(loadingPlace).appendChild(gif);
+};
+
+const removeLoading = () => {
+    document.getElementById("loadingPlace").remove();
+};
+
+const appendListForParent = (value) => {
+    const ul = document.createElement('ul');
+    const fragment = document.createDocumentFragment();
+    
+    value.data.forEach(key => {
```

Comment:
IMO
```suggestion
    value.data.forEach(d => {
```
It's better to shorten it if it has a different meaning.
you can't get the key even if I turn data with forEach.

```js
["a", "b", "c"]
```
you just get element or value or item. not key

Reply:
IMO

If I use `key` here, I would get json's data in the `getData` first.

Like this↓
```
const getData = async () => {
  const response = await fetch(
    'https://jsondata.okiba.me/v1/json/3jhPW210812052923'
  );
  const json = await response.json();
  return json.data;
};
```

Reply:
Thank you for your advice.
I learned a lot. I would like to utilize it next time without modifying it this time.